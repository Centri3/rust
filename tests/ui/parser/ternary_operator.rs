#![no_main]

fn a() {
    let x = 5 > 2 ? true : false; //~ ERROR Rust has no ternary operator
    //~^ HELP use an `if-else` expression instead
    //~^^ ERROR the `?` operator can only be applied to values that implement `Try`
    //~^^^ HELP the trait `Try` is not implemented for `{integer}`
    //~^^^^ ERROR the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`) [E0277]
    //~^^^^^ HELP the trait `FromResidual<_>` is not implemented for `()`
}

fn b() {
    let x = 5 > 2 ? { true } : { false }; //~ ERROR Rust has no ternary operator
    //~^ HELP use an `if-else` expression instead
    //~^^ ERROR the `?` operator can only be applied to values that implement `Try`
    //~^^^ HELP the trait `Try` is not implemented for `{integer}`
    //~^^^^ ERROR the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`) [E0277]
    //~^^^^^ HELP the trait `FromResidual<_>` is not implemented for `()`
}

fn c() {
    let x = 5 > 2 ? f32::MAX : f32::MIN; //~ ERROR Rust has no ternary operator
    //~^ HELP use an `if-else` expression instead
    //~^^ ERROR the `?` operator can only be applied to values that implement `Try`
    //~^^^ HELP the trait `Try` is not implemented for `{integer}`
    //~^^^^ ERROR the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`) [E0277]
    //~^^^^^ HELP the trait `FromResidual<_>` is not implemented for `()`
}
