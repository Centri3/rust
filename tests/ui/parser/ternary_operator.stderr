error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:4:19
   |
LL |     let x = 5 > 2 ? true : false;
   |                   ^
   |
   = help: use an `if-else` expression instead

error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:13:19
   |
LL |     let x = 5 > 2 ? { true } : { false };
   |                   ^
   |
   = help: use an `if-else` expression instead

error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:22:19
   |
LL |     let x = 5 > 2 ? f32::MAX : f32::MIN;
   |                   ^
   |
   = help: use an `if-else` expression instead

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:4:17
   |
LL |     let x = 5 > 2 ? true : false;
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
  --> $DIR/ternary_operator.rs:4:19
   |
LL | fn a() {
   | ------ this function should return `Result` or `Option` to accept `?`
LL |     let x = 5 > 2 ? true : false;
   |                   ^ cannot use the `?` operator in a function that returns `()`
   |
   = help: the trait `FromResidual<_>` is not implemented for `()`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:13:17
   |
LL |     let x = 5 > 2 ? { true } : { false };
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
  --> $DIR/ternary_operator.rs:13:19
   |
LL | fn b() {
   | ------ this function should return `Result` or `Option` to accept `?`
LL |     let x = 5 > 2 ? { true } : { false };
   |                   ^ cannot use the `?` operator in a function that returns `()`
   |
   = help: the trait `FromResidual<_>` is not implemented for `()`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:22:17
   |
LL |     let x = 5 > 2 ? f32::MAX : f32::MIN;
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
  --> $DIR/ternary_operator.rs:22:19
   |
LL | fn c() {
   | ------ this function should return `Result` or `Option` to accept `?`
LL |     let x = 5 > 2 ? f32::MAX : f32::MIN;
   |                   ^ cannot use the `?` operator in a function that returns `()`
   |
   = help: the trait `FromResidual<_>` is not implemented for `()`

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0277`.
